File: 153. Find Minimum in Rotated Sorted Array/153.java
// 153. Find Minimum in Rotated Sorted Array solution in Java
// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/

class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            int mid = left + ((right - left) / 2);

            if (nums[mid] > nums[right]) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return nums[left];
    }
}

File: 153. Find Minimum in Rotated Sorted Array/153.go
// 153. Find Minimum in Rotated Sorted Array solution in Golang
// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/

package main

func findMin(nums []int) int {
	left := 0
	right := len(nums) - 1

	for left < right {
		mid := left + ((right - left) / 2)

		if nums[mid] > nums[right] {
			left = mid + 1
		} else {
			right = mid
		}
	}

	return nums[left]
}
